x/ibc/02-client/types/msgs.go

```diff
@@ -175,12 +173,71 @@ func (msg MsgUpdateClient) GetSigners() []sdk.AccAddress {
 // UnpackInterfaces implements UnpackInterfacesMessage.UnpackInterfaces
 func (msg MsgUpdateClient) UnpackInterfaces(unpacker codectypes.AnyUnpacker) error {
 	var header exported.Header
-	err := unpacker.UnpackAny(msg.Header, &header)
+	return unpacker.UnpackAny(msg.Header, &header)
+}
+
+// NewMsgUpgradeClient creates a new MsgUpgradeClient instance
+// nolint: interfacer
+func NewMsgUpgradeClient(clientID string, clientState exported.ClientState, proofUpgrade []byte, signer sdk.AccAddress) (*MsgUpgradeClient, error) {
+	anyClient, err := PackClientState(clientState)
+	if err != nil {
+		return nil, err
+	}
+
+	return &MsgUpgradeClient{
+		ClientId:     clientID,
+		ClientState:  anyClient,
+		ProofUpgrade: proofUpgrade,
+		Signer:       signer.String(),
+	}, nil
+}
+
+// Route implements sdk.Msg
+func (msg MsgUpgradeClient) Route() string {
+	return host.RouterKey
+}
+
+// Type implements sdk.Msg
+func (msg MsgUpgradeClient) Type() string {
+	return TypeMsgUpgradeClient
+}
+
+// ValidateBasic implements sdk.Msg
+func (msg MsgUpgradeClient) ValidateBasic() error {
+	clientState, err := UnpackClientState(msg.ClientState)
 	if err != nil {
 		return err
 	}
+	if err := clientState.Validate(); err != nil {
+		return err
+	}
+	if len(msg.ProofUpgrade) == 0 {
+		return sdkerrors.Wrap(ErrInvalidUpgradeClient, "proof of upgrade cannot be empty")
+	}
+	if msg.Signer == "" {
```

Comment:
use `strings.TrimSpace`

Reply:
I'd also check that `accAddr, err := sdk.AccAddressFromBech32(msg.Signer)` doesn't error, just to make sure that the `GetSigners` func won't panic